### PR TITLE
SCANMAVEN-220: Enable sonar.scanner.scanAll by default

### PR DESCRIPTION
**Pull request by ADarko22**
Please be aware that we are not actively looking for feature contributions. The truth is that it's extremely difficult for someone outside SonarSource to comply with our roadmap and expectations. Therefore, we typically only accept minor cosmetic changes and typo fixes. If you would like to see a new feature, please create a new thread in the forum ["Suggest new features"](https://community.sonarsource.com/c/suggestions/features).

With that in mind, if you would like to submit a code contribution, make sure that you adhere to the following guidelines and all tests are passing:

- [ ] Please explain your motives to contribute this change: what problem you are trying to fix, what improvement you are trying to make
- [ ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [ ] Provide a unit test for any code you changed
- [ ] If there is a [JIRA](http://jira.sonarsource.com/browse/MSONAR) ticket available, please make your commits and pull request start with the ticket ID (MSONAR-XXXX)

We will try to give you feedback on your contribution as quickly as possible.

Thank You!
The SonarSource Team

### CODE REVIEW

File: src/it/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Sonar Scanner Maven Plugin Integration Tests
```
Comment (leveretka) [inline src/it/README.md:1]: I think we already have some documentation about ITs here: https://github.com/SonarSource/sonar-scanner-maven/blob/master/develop.md

What do you think if we merge these two documents, or move all the information about ITs here?

File: src/test/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapperTest.java [hunk 2]
```diff
@@ -333,6 +328,19 @@ void maven_opts_is_not_logged_at_info_level_when_not_absent_from_environment_var
     }
   }
 
+  private void setSonarScannerScanAllTo(String value) {
+    Properties withScanAllSet = new Properties();
+    withScanAllSet.put(MavenScannerProperties.PROJECT_SCAN_ALL_SOURCES, value);
+    when(session.getUserProperties()).thenReturn(withScanAllSet);
+  }
+
+  private void verifyCollectedSources(Consumer<String[]> sourceDirsAssertions) throws MojoExecutionException {
```
Comment (leveretka) [inline src/test/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapperTest.java:337]: NIce idea with these methods, thanks for simplifying testing.

**Review by leveretka — APPROVED** [review]
A minor suggestion, otherwise looks good too me!

File: src/it/README.md (hunk 1 above)
Comment (ADarko22) [reply src/it/README.md]: Oh nice! I will remove it then :)

File: src/test/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapperTest.java (hunk 2 above)
Comment (ADarko22) [reply src/test/java/org/sonarsource/scanner/maven/bootstrap/ScannerBootstrapperTest.java:337]: Thank you :)

**Pull request merged**